### PR TITLE
Add *WithClaims methods to jwt middleware for more advanced usage.

### PR DESCRIPTION
**Pull request by cam-stitt**
I started migrating our existing JWT layer to go-kit's version however I was not able to complete the process. We implement a custom jwt.Claims object and need to be able to add middleware layers that use this.

This PR adds some new exported middlewares that allow for advanced claim types.

### CODE REVIEW

**Comment by cam-stitt** [comment]
I'm happy with the state of this PR. Would appreciate a review :+1:

**Comment by peterbourgon** [comment]
I don't have enough context or knowledge of JWT to properly review the semantics here, but the code looks good. Unless @briankassouf wants to object, I'll merge in 48h.

File: auth/jwt/middleware_test.go [hunk 1]
```diff
@@ -94,12 +104,26 @@ func TestJWTParser(t *testing.T) {
...
+	ctx1, err = parser(ctx, struct{}{})
+	if err != nil {
+		t.Fatalf("Parser returned error: %s", err)
+	}
+	stdCl, ok := ctx1.(context.Context).Value(JWTClaimsContextKey).(*jwt.StandardClaims)
+	if !ok {
+		t.Fatal("Claims were not passed into context correctly")
+	}
+	if !stdCl.VerifyAudience("go-kit", true) {
+		t.Fatal("JWT jwt.StandardClaims.Audience did not match: expecting %s got %s", standardClaims.Audience, stdCl.Audience)
```
Comment (basvanbeek) [inline auth/jwt/middleware_test.go]: `t.Fatalf`

**Comment by basvanbeek** [comment]
Only one little `t.Fatalf` but for the rest LGTM.

I do have one potential addition for the middleware_test in highlighting the typical practice of embedding a `jwt.StandardClaims` struct to create a custom claims type.

I've taken the liberty to add onto this PR and code can be seen here:

https://github.com/basvanbeek/kit/blob/jwt-claims/auth/jwt/middleware_test.go

**Comment by briankassouf** [comment]
This change looks good to me!

**Comment by peterbourgon** [comment]
Thanks @cam-stitt and everyone!

**Pull request merged**